File: Data Structures & Algorithms/sum-of-two-integers/submission-0.java
public class Solution {
    public int getSum(int a, int b) {
        // Keep looping until there is no carry left
        while (b != 0) {
            // Carry is calculated by bitwise AND and then left-shifted by 1
            // because a carry affects the next higher bit
            int carry = (a & b) << 1;

            // Sum without carrying is done using bitwise XOR
            a ^= b;

            // Assign carry to b so that it will be added in the next iteration
            b = carry;
        }

        // When carry becomes zero, a contains the result
        return a;
    }
}

File: Data Structures & Algorithms/missing-number/submission-0.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        sum_ = (n * (n + 1) ) / 2

        sum_nums = sum(nums)
        print(sum_,sum_nums)

        if sum_ == sum_nums:
            return 0
        else:
            return int(sum_ - sum_nums)
